_specs/51_attack_resistance.md

```diff
@@ -0,0 +1,282 @@
+
+This list can be publicly accessed through a getter smart-contract.
+
+This method can then be called by the nodes that participate in the checkpointing network, and only the nodes listed can issue checkpoints themselves.
+
+- Pros:
+    - Community-driven and on-chain selection of com mittee
+    - Reliability of the n participants
+
+- Cons:
+    - A very wealthy attacker can register multiple public keys and vote for himself to obtain a majority of checkpoint signatures 
+    - This is an example of how one could implement a voting scheme, however this scheme starts to resemble Proof of Stake and it may be worthwhile considering hybrid consensus if this scheme is deemed favourable by the community
+
+
+##### Option 2: Dynamic network of ETC miners
+
+In this dynamic and random selection option, the checkpointing network is operated and secured by a committee of k miners who are randomly selected at every epoch through the blockchain data structure and history. The parameters for this random selection can be either :
+- based on the lexical distance of the miner’s public key with the last block hash;
+- within a window of time determining a subset of miners;
+- within a set of NiPoPoWs [^NIPOPOW] graphs;
+
+We envisage that a multitude of randomized selection algorithms could be used, but we advise the community to select the solution that offers the best resistance to attacks.
+
+The specific implementation of this dynamic and random selection option will be provided as a consensus algorithm used in the checkpointing blockchain that actively monitors the ETC blockchain. The following is a draft proposal for the committee selection algorithm that we consider offers  sufficient protection against long range attacks:
+- `e` an epoch of a duration t in which a new list of at least `n` miners is determined;
+- `p` the maximum processing time for this algorithm;
+- `w` a sliding window of `k blocks` from the latest block of the latest epoch `window + window_min` (or `last block - window_min` at initialization), `k` calculated based on the difficulty  and  `k > window_min`;
+- calculate `npp(k)` the subset of `k` latest NiPoPow graphs;
+- calculate `l(npp(k))`the list of miners’ public keys that have contributed to mine the blocks contained by `npp(k)`removing duplicates (set);
+- if `length(l(npp(k))) < n` go to 2.;
+- else the list is determined 
+- when `t - p` expires go to 1.
+
+- Pros
+    - Checkpointing network is secured by miners
+    - On-chain selection of committee
+    - Decreased risk of collusion within the committee
+- Cons
+    - Sensitive to long range attacks 
+    - Draft solution and the secure algorithm, or equivalent, needs to be formalised
+    - It may not be possible to prove a satisfactory level of security with this scheme, also to make such a scheme secure will likely increase implementation complexity
+
+##### Option 3: Auxiliary Security from an External Blockchain
```

Comment:
What does an attacker prevent from creating valid timestamping transactions on Bitcoin?

Reply:
Timestamping does not try to prevent anyone from timestamping blocks, it only ensures that a single chain is (irreversibly) accepted as canonical. Once the first block of a specific height is timestamped, then it acts as a checkpoint (regardless of who created it) and no party that follows the timestamping mechanism will revert it in the future (thus preventing deep chain reorgs). In other words, allowing an attacker to timestamp blocks is acceptable under this model; what is not acceptable is reorganizing a block that has been timestamped and has been adopted as the local chain.